core/src/main/scala/org/bitcoins/core/protocol/dlc/DLCTemplate.scala

```diff
@@ -0,0 +1,296 @@
+package org.bitcoins.core.protocol.dlc
+
+import org.bitcoins.core.currency._
+import org.bitcoins.core.protocol.dlc.ContractOraclePair._
+import org.bitcoins.core.protocol.dlc.DLCTemplate.validateMatchingOracleDescriptors
+import org.bitcoins.core.protocol.tlv._
+
+sealed trait DLCTemplate {
+  def totalCollateral: CurrencyUnit
+
+  def oracles: Vector[OracleAnnouncementTLV]
+
+  def oracleThreshold: Int
+
+  def contractDescriptor: ContractDescriptor
+
+  def oracleInfo: OracleInfo
+
+  def toContractInfo: ContractInfo
```

Comment:
I think everything here except for `individualCollateral` can be defined from this right? Either that or the other way around (meaning this can be defined from stuff here)

Reply:
Unsure what you mean here. We want to produce a `ContractInfo` from the template, so we should be entering params to construct one.